modules/hal_rpi_pico/pico/config_autogen.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021, Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Originally this file is generated by pico-sdk, and some files
+ * try to include it. Therefore, we have to provide that file,
+ * with this exact name.
+ * Since this file ends up included in all pico-sdk code, it's
+ * used to inject workarounds to make pico-sdk compile with Zephyr.
+ */
+
+#ifndef _CONFIG_AUTOGEN_H_
+#define _CONFIG_AUTOGEN_H_
+
+/* WORKAROUNDS */
+
+/* Use Zephyr's __asm macro instead of pico-sdk's asm */
+#define asm __asm
```

Comment:
If I understand this correctly, `asm` is not supported with `-std=c99`, so this macro just converts the few instances of `asm` in pico-sdk to the supported format (I should fix the comment above). I couldn't find anything in Zephyr that would do the same, and I prefer not to include extra headers to this file, as it can cause other problems similar to the KHZ/MHZ thing.